### ISSUE TITLE
suggestion; disable hostile mob spawns in unfinished structures

### ISSUE THREAD

**Issue by adam110902** [ldtteam/minecolonies-features#854]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my feature does not already exist.
- [x] I made sure that **this feature request is not a duplicate of any existing feature request.**

### Context 

- Minecraft Version:1.20.1
- MineColonies Version:1.1.686

## Is your feature request related to a problem?
hostile mobs sometimes spawn in unfinished buildings

## Feature description
some way to prevent hostile mobs from spawning during building, maybe use light blocks set to low to make it less noticable 

### Notes or related things
it could be locked behind research and/or exclusive to larger structures

---
## Viewers

* Add a thumbs-up to the feature request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Nitka711** [comment]
That's a great idea. This has bothered me for some time now. My builders are getting killed in their builds, because they are dark inside.

**Comment by Nitka711** [comment]
I circumvented this problem at least for upgrades by having torches for tier 1 and leaving them for tier 2 but adding lanterns. Only at tier 3 the torches are removed, so as to preventing any length of time without light.
Of course this does not change anything until tier 1 is build, as the builder first digs everything he needs out. With any underground building or nighttime there will be a problem with mobs spawning, as the builder removes everything (torches etc) before starting to build.
So something like a prevention from spawning hostile mobs in a started build would be great..... Personally I would like it if Hostile mobs could not spawn in the whole colonie.

**Comment by aviemet** [comment]
Oh man, the number of builders I've lost because they excavated and ended up in a light level 0 area. I think another simple solution would be to have the builder place torches if they're ever in a light level 0 area. They could be told to always place light emitting blocks first when building, that way breaking torches wouldn't lead to an infinite loop. I also like just disabling spawns in unfinished builds.
